### PR TITLE
Add support for model objects with RealmUUID

### PR DESCRIPTION
**Pull request by clementetb**
Part of https://github.com/realm/realm-kotlin/pull/891

This PR adds support for modeling objects with `RealmUUID` fields:

- [x] `UUID` `RealmObject` fields
- [x] `UUID` `RealmList`
- [x] Queries

### CODE REVIEW

File: packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt [hunk 1]
```diff
@@ -1503,6 +1514,17 @@ private fun capiRealmValue(realmValue: RealmValue): realm_value_t {
                 cvalue.link = realmc.realm_object_as_link(nativePointer.cptr())
                 cvalue.type = realm_value_type_e.RLM_TYPE_LINK
             }
+            is UUIDWrapper -> {
+                cvalue.type = realm_value_type_e.RLM_TYPE_UUID
+                cvalue.uuid = realm_uuid_t().apply {
+                    val data = ShortArray(UUID_BYTES_SIZE)
+                    @OptIn(ExperimentalUnsignedTypes::class)
```
Comment (nhachicha) [inline packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt:1521]: redundant

**Review by nhachicha — APPROVED** [review]
💯

File: packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/Converters.kt [hunk 2]
```diff
@@ -164,6 +174,8 @@ internal val primitiveTypeConverters: Map<KClass<*>, RealmValueConverter<*>> =
         Int::class to IntConverter,
         RealmInstant::class to RealmInstantConverter,
         ObjectId::class to ObjectIdConverter,
+        RealmUUID::class to RealmUUIDConverter,
+        ObjectId::class to ObjectIdConverter,
```
Comment (rorbech) [inline packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/Converters.kt:178]:
```suggestion
```

**Review by rorbech — APPROVED** [review]
LGTM - Only some minor comments here and there

File: packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt [hunk 3]
```diff
@@ -1507,6 +1518,17 @@ private fun capiRealmValue(realmValue: RealmValue): realm_value_t {
...
+                    (0 until UUID_BYTES_SIZE).map {
+                        data[it] = value.bytes[it].toShort()
+                    }
+                    bytes = data
```
[the remark this answers is not in the document]
Comment (clementetb) [reply packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt:1528]: It seems it does.

**Pull request merged**